### PR TITLE
Add API v3 routes for compitable with github

### PR DESCRIPTION
**Pull request by lunny**
Copy all v1 routes to v3 currently.

### CODE REVIEW

**Comment by lafriks** [comment]
Can you add in the way that only one can be changed?

**Comment by lunny** [comment]
@lafriks they point the same route function currently. So you will always change both.

**Comment by lafriks** [comment]
@lunny there is PR to change api route name to match github (#2639) so it would be nice to be able to do that only for v3 and for v1 leave as it is to not break anything just in case

File: routers/api/v1/api.go [hunk 1]
```diff
@@ -266,287 +266,295 @@ func mustAllowPulls(ctx *context.Context) {
...
+// FIXME: custom form error response
+func RegisterRoutes(m *macaron.Macaron) {
+	m.Get("/swagger", misc.Swagger) //Render V1 by default
+
+	m.Group("/v1", func() {
+		apiRoutes(m)
+	}, context.APIContexter())
+
+	m.Group("/v3", func() {
+		apiRoutes(m)
```
Comment (bkcsoft) [inline routers/api/v1/api.go:558]: Some of these routes are not GitHub-compatible. They shouldn't be added to `v3/` 😕

**Review by bkcsoft — CHANGES_REQUESTED** [review]

**Comment by lunny** [comment]
OK. I think we could add new route `/v3` and only move github api v3 comiptable routes there.

**Comment by sapk** [comment]
It would be best to also separate the route into sub-route to not have a "big file" like v1.

File: routers/api/v1/api.go [hunk 2]
```diff
@@ -266,287 +266,295 @@ func mustAllowPulls(ctx *context.Context) {
 	}
 }
 
-// RegisterRoutes registers all v1 APIs routes to web application.
-// FIXME: custom form error response
-func RegisterRoutes(m *macaron.Macaron) {
+func apiRoutes(m *macaron.Macaron) {
```
Comment (bkcsoft) [inline routers/api/v1/api.go:269]: I would add a `ghCompat bool` argument here, and `if !ghCompat { ... }` around any custom endpoints we have

**Comment by techknowlogick** [comment]
Closing per above conversation, we can open a fresh PR that includes only routes compatible with GitHub v3

**Pull request closed without merging**